src/projected.jl

```diff
@@ -0,0 +1,54 @@
+abstract type AbstractProjectedCoords <: AbstractSkyCoords end
+
+""" Projected coordinates with origin and offset.
+Approximation for small FoV. """
+struct ProjectedCoords{TC<:AbstractSkyCoords,T} <: AbstractProjectedCoords
+    origin::TC
+    offset::SVector{2,T}
+end
+
+""" Projected coordinates with origin (stored in the type) and offset.
+Approximation for small FoV. """
+struct ProjectedCoordsS{C,T} <: AbstractProjectedCoords
+    offset::SVector{2,T}
+end
+
+ProjectedCoordsS{C}(offset) where {C} = ProjectedCoordsS{C, eltype(offset)}(offset)
+constructorof(::Type{<:ProjectedCoordsS{C}}) where {C} = ProjectedCoordsS{C} ∘ tuple
+
+parent_coords_type(::Type{ProjectedCoords}) = Any
+parent_coords_type(::Type{ProjectedCoordsS}) = Any
+parent_coords_type(::Type{<:ProjectedCoords{TC}}) where {TC} = TC
+parent_coords_type(::Type{<:ProjectedCoordsS{TC}}) where {TC <: AbstractSkyCoords} = TC
+parent_coords_type(::Type{<:ProjectedCoordsS{origin}}) where {origin} = typeof(origin)
+
+origin(c::ProjectedCoords) = c.origin
+origin(::ProjectedCoordsS{o}) where {o} = o
+
+Base.convert(::Type{T}, c::T) where {T<:AbstractProjectedCoords} = c
+Base.convert(::Type{ProjectedCoords}, c::ProjectedCoordsS) = ProjectedCoords(origin(c), c.offset)
+Base.convert(::Type{ProjectedCoordsS}, c::ProjectedCoords) = ProjectedCoordsS{origin(c)}(c.offset)
+
+function Base.convert(::Type{TCto}, c::AbstractProjectedCoords) where {TCto<:AbstractSkyCoords}
+    o = origin(c)
+    res = constructorof(typeof(o))(lon(o) + c.offset[1] / cos(lat(origin(c))), lat(o) + c.offset[2])
+    convert(TCto, res)
+end
+
+Base.isapprox(a::ProjectedCoords, b::ProjectedCoords; kwargs...) = isapprox(origin(a), origin(b); kwargs...) && isapprox(a.offset, b.offset; kwargs...)
+Base.isapprox(a::ProjectedCoordsS{C}, b::ProjectedCoordsS{C}; kwargs...) where {C} = isapprox(a.offset, b.offset; kwargs...)
+
+
+project(::Val{origin}, c::AbstractSkyCoords) where {origin} = ProjectedCoordsS{origin}(project(origin, c).offset)
+
+function project(origin::AbstractSkyCoords, c::AbstractSkyCoords)
+    cc = convert(typeof(origin), c)
+    Δlon = _center_angle(lon(cc) - lon(origin))
+    offset = SVector(Δlon * cos(lat(origin)), lat(cc) - lat(origin))
+    ProjectedCoords(origin, offset)
+end
+
+_center_angle(x) = mod(x + π, 2π) - π
```

Comment:
This only appears to be used in one place, is it necessary to define this method?